pkg/scheduler/flavorassigner/tas_flavorassigner.go

```diff
@@ -20,64 +20,62 @@ import (
-	case len(cq.TASFlavors) == 0:
-		if psAssignment.Status == nil {
-			psAssignment.Status = &Status{}
-		}
-		psAssignment.Status.append("Workload requires Topology, but there is no TAS cache information")
-		psAssignment.Flavors = nil
-	default:
-		singlePodRequests := psResources.Requests.Clone()
-		singlePodRequests.Divide(int64(psResources.Count))
-		podCount := psAssignment.Count
-		tasFlvr, err := onlyFlavor(psAssignment.Flavors)
-		if err != nil {
-			if psAssignment.Status == nil {
-				psAssignment.Status = &Status{}
-			}
-			psAssignment.Status.err = err
-			psAssignment.Flavors = nil
-			return
-		}
-		snapshot := cq.TASFlavors[*tasFlvr]
-		if snapshot == nil {
-			if psAssignment.Status == nil {
-				psAssignment.Status = &Status{}
+// WorkloadsTopologyRequests - returns the TopologyRequests of the workload
+func (a *Assignment) WorkloadsTopologyRequests(wl *workload.Info,
+	requests []workload.PodSetResources,
+	cq *cache.ClusterQueueSnapshot) cache.WorkloadTASRequests {
+	tasRequests := make(cache.WorkloadTASRequests)
+	for i, podSet := range wl.Obj.Spec.PodSets {
+		if podSet.TopologyRequest != nil {
+			psAssignment := a.podSetAssignmentByName(podSet.Name)
+			if psAssignment.Status.IsError() {
+				// There is no resource quota assignment for the PodSet - no need to check TAS.
+				continue
 			}
-			psAssignment.Status.append("Workload requires Topology, but there is no TAS cache information for the assigned flavor")
-			psAssignment.Flavors = nil
-			return
-		}
-		var reason string
-		psAssignment.TopologyAssignment, reason = snapshot.FindTopologyAssignment(podSet.TopologyRequest,
-			singlePodRequests, podCount, podSet.Template.Spec.Tolerations)
-		if psAssignment.TopologyAssignment == nil {
-			if psAssignment.Status == nil {
-				psAssignment.Status = &Status{}
+			psTASRequest, err := podSetTopologyRequest(psAssignment, wl, cq, i)
+			if err != nil {
+				psAssignment.error(err)
+			} else {
+				tasRequests[psTASRequest.Flavor] = append(tasRequests[psTASRequest.Flavor], *psTASRequest)
 			}
-			psAssignment.Status.append(reason)
-			psAssignment.Flavors = nil
 		}
-		log.Info("TAS PodSet assignment", "tasAssignment", psAssignment.TopologyAssignment)
 	}
+	return tasRequests
+}
+
+func podSetTopologyRequest(psAssignment *PodSetAssignment,
+	wl *workload.Info,
+	cq *cache.ClusterQueueSnapshot,
+	podSetIndex int) (*cache.TASPodSetRequests, error) {
+	if len(cq.TASFlavors) == 0 {
+		return nil, errors.New("workload requires Topology, but there is no TAS cache information")
+	}
+	psResources := wl.TotalRequests[podSetIndex]
+	singlePodRequests := psResources.Requests.Clone()
+	singlePodRequests.Divide(int64(psResources.Count))
+	podCount := psAssignment.Count
+	tasFlvr, err := onlyFlavor(psAssignment.Flavors)
+	if err != nil {
+		return nil, err
+	}
+	snapshot := cq.TASFlavors[*tasFlvr]
+	if snapshot == nil {
```

Comment:
```suggestion
	if cq.TASFlavors[*tasFlvr] == nil {
```

I think this can avoid allocation.

Reply:
Or, we might be able to just check if `cq.TASFlavors` has `tasFlvr` key like `if _, ok := ...; !ok`.
Is there any situation where the TASFlavors has key, but value is null?

Reply:
Updated.
> Is there any situation where the TASFlavors has key, but value is null?

This should not happen,